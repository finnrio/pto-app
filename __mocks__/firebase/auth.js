// __mocks__/firebase/auth.js
export const getAuth = jest.fn(() => {
  return {
    currentUser: { uid: 'testUid' },
    signInWithEmailAndPassword: jest.fn(),
    signOut: jest.fn(),
  };
});
