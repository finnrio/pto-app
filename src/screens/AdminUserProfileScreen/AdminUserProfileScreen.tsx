import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import { Alert, Text, TextInput, TouchableOpacity, View } from "react-native";
import React, { useEffect, useState } from "react";
import { SegmentedButtons } from "react-native-paper";
import DropDownPicker from "react-native-dropdown-picker";

import UpdateUserData from "../../firebase/operations/UpdateUserData";
import styles from "./styles";
import { AppUser } from "../../types/AppUser";
import GetAllManagers from "../../firebase/operations/GetAllManagers";
import GetAllUsers from "../../firebase/operations/GetAllUsers";
import DeleteUser from "../../firebase/operations/DeleteUser";
import GetUserData from "../../firebase/operations/GetUserData";

export default function AdminUserProfileScreen() {
  const [firstName, setFirstName] = useState("");
  const [surname, setSurname] = useState("");
  const [email, setEmail] = useState("");
  const [role, setRole] = useState("");
  const [manager, setManager] = useState("");
  const [managerList, setManagerList] = useState<any[]>([]);
  const [userList, setUserList] = useState<any[]>([]);
  const [ptoUsed, setPTOUsed] = useState("");
  const [ptoAllowance, setPTOAllowance] = useState("");
  const [allUserList, setAllUserList] = useState<any[]>([]);
  const [selectedUser, setSelectedUser] = useState("");
  const [openUser, setOpenUser] = useState(false);
  const [openManager, setOpenManager] = useState(false);
  const [scrollEnabled, setScrollEnabled] = useState(true);

  async function RenderUserData(id: string) {
    const res: AppUser = await GetUserData(id);
    setFirstName(res?.first_name);
    setSurname(res?.surname);
    setEmail(res?.email);
    setRole(res?.role);
    setManager(res?.manager_id ? res.manager_id : "");
    setPTOUsed(res?.pto_used);
    setPTOAllowance(res?.pto_allowance);
  }

  async function RenderUserLists() {
    await GetAllUsers()
      .then((response) => setUserList(response))
      .catch((e) => Alert.alert(e.code, e.message));
    await GetAllManagers()
      .then((response) => setManagerList(response))
      .catch((e) => Alert.alert(e.code, e.message));
  }

  function createUserDataObject() {
    return {
      first_name: firstName,
      surname,
      role,
      manager_id: manager,
      pto_allowance: ptoAllowance,
      pto_used: ptoUsed,
    };
  }

  function HandleUpdateBtn() {
    try {
      Alert.alert(
        "Update users profile details",
        "Are you sure you want to update your profile details?",
        [
          {
            text: "Update",
            onPress: () => {
              UpdateUserData(createUserDataObject(), selectedUser);
              RenderUserData("");
            },
            style: "default",
          },
          {
            text: "Reset Changes",
            onPress: () => {
              RenderUserData(selectedUser);
              RenderUserLists();
            },
            style: "cancel",
          },
          {
            text: "Cancel",
            style: "cancel",
          },
        ],
        { cancelable: true },
      );
    } catch (e: unknown) {
      console.error(e);
      if (e instanceof Error) {
        Alert.alert(e.name, e.message);
      } else {
        Alert.alert("Error", "An error has occured");
      }
    }
  }

  function HandleRemoveUserBtn() {
    Alert.alert(
      "Warning",
      `Are you sure you want to delete ${firstName} ${surname} from the system?`,
      [
        {
          text: "Delete",
          onPress: () => {
            DeleteUser(selectedUser);
            setSelectedUser("");
            RenderUserData(selectedUser);
          },
          style: "default",
        },
        {
          text: "Cancel",
          style: "cancel",
        },
      ],
      { cancelable: true },
    );
  }

  useEffect(() => {
    RenderUserLists();
  }, []);

  useEffect(() => {
    setAllUserList(userList.concat(managerList));
  }, [userList, managerList]);

  useEffect(() => {
    if (selectedUser) RenderUserData(selectedUser);
    RenderUserLists();
  }, [selectedUser]);

  return (
    <View style={styles.container}>
      <KeyboardAwareScrollView
        style={{ flex: 1, width: "100%" }}
        keyboardShouldPersistTaps="always"
        scrollEnabled={scrollEnabled}
      >
        <DropDownPicker
          open={openUser}
          value={selectedUser}
          items={allUserList}
          setOpen={setOpenUser}
          setValue={setSelectedUser}
          setItems={setAllUserList}
          onOpen={() => setScrollEnabled(false)}
          onClose={() => setScrollEnabled(true)}
          searchable={true}
          testID="user_dropdown"
        />
        <Text style={styles.text}>User ID</Text>
        <TextInput
          style={styles.input}
          placeholder="No user selected"
          placeholderTextColor="#aaaaaa"
          value={selectedUser}
          underlineColorAndroid="transparent"
          editable={false}
          testID="uid_input"
        />
        <Text style={styles.text}>First Name</Text>
        <TextInput
          style={styles.input}
          placeholder="First Name"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => {
            setFirstName(text);
          }}
          value={firstName}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
          editable={!!selectedUser}
          testID="firstName_input"
        />
        <Text style={styles.text}>Surname</Text>
        <TextInput
          style={styles.input}
          placeholder="Surname"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => {
            setSurname(text);
          }}
          value={surname}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
          editable={!!selectedUser}
          testID="surname_input"
        />
        <Text style={styles.text}>Email</Text>
        <TextInput
          style={styles.input}
          placeholder="E-mail"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => {
            setEmail(text);
          }}
          value={email!}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
          editable={false}
          testID="email_input"
        />
        <Text style={styles.text}>Role</Text>
        <SegmentedButtons
          value={role}
          onValueChange={setRole}
          buttons={[
            {
              value: "User",
              label: "User",
              disabled: !selectedUser,
              testID: "user_button",
            },
            {
              value: "Manager",
              label: "Manager",
              disabled: !selectedUser,
              testID: "manager_button",
            },
          ]}
        />
        <Text style={styles.text}>Manager</Text>
        <DropDownPicker
          open={openManager}
          value={manager}
          items={managerList}
          setOpen={setOpenManager}
          setValue={setManager}
          setItems={setManagerList}
          onOpen={() => setScrollEnabled(false)}
          onClose={() => setScrollEnabled(true)}
          searchable={true}
          testID="manager_dropdown"
        />
        <Text style={styles.text}>PTO Allowance</Text>
        <TextInput
          style={styles.input}
          placeholder="PTO Allowance"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => {
            setPTOAllowance(text);
          }}
          value={ptoAllowance?.toString()}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
          editable={!!selectedUser}
          testID="pto_allowance_input"
        />
        <Text style={styles.text}>PTO Used</Text>
        <TextInput
          style={styles.input}
          placeholder="PTO Used"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => {
            setPTOUsed(text);
          }}
          value={ptoUsed?.toString()}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
          editable={!!selectedUser}
          testID="pto_used_input"
        />
        <TouchableOpacity
          style={selectedUser ? styles.button : styles.buttonUnavailable}
          onPress={() =>
            selectedUser ? HandleUpdateBtn() : Alert.alert("No user selected")
          }
          testID="update_button"
        >
          <Text style={styles.buttonTitle}>Update Profile</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={selectedUser ? styles.button : styles.buttonUnavailable}
          onPress={() =>
            selectedUser
              ? HandleRemoveUserBtn()
              : Alert.alert("No user selected")
          }
          testID="remove_button"
        >
          <Text style={styles.buttonTitle}>Remove User</Text>
        </TouchableOpacity>
      </KeyboardAwareScrollView>
    </View>
  );
}
