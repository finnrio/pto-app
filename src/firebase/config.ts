import { initializeApp } from 'firebase/app';
import { getAuth } from "firebase/auth";
import { getFirestore } from "firebase/firestore";

// Optionally import the services that you want to use
// import {...} from "firebase/auth";
// import {...} from "firebase/database";
// import {...} from "firebase/firestore";
// import {...} from "firebase/functions";
// import {...} from "firebase/storage";

// Initialize Firebase
const firebaseConfig = {
  apiKey: 'AIzaSyBEjW_646bH_Q4ECQD_g7nXhM_WTDQQZqw',
  authDomain: 'pto-app-30271.firebaseapp.com',
  databaseURL: 'https://pto-app-30271.firebaseio.com',
  projectId: 'pto-app-30271',
  storageBucket: 'pto-app-30271.appspot.com',
  messagingSenderId: '440515616929',
  appId: '1:440515616929:ios:1be65de5cfa7a0048210e0',
  measurementId: '1:440515616929:ios:1be65de5cfa7a0048210e0',
};

export const FIREBASE_APP = initializeApp(firebaseConfig);
// For more information on how to access Firebase in your project,
// see the Firebase documentation: https://firebase.google.com/docs/web/setup#access-firebase

export const FIREBASE_AUTH = getAuth(FIREBASE_APP);
export const FIRESTORE_DB = getFirestore(FIREBASE_APP);