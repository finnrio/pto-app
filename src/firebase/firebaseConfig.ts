import { initializeApp } from "firebase/app";
import { connectAuthEmulator, getAuth } from "firebase/auth";
import { connectFirestoreEmulator, getFirestore } from "firebase/firestore";

// Initialize Firebase
const firebaseConfig = {
  apiKey: "AIzaSyBEjW_646bH_Q4ECQD_g7nXhM_WTDQQZqw",
  authDomain: "pto-app-30271.firebaseapp.com",
  databaseURL: "https://pto-app-30271.firebaseio.com",
  projectId: "pto-app-30271",
  storageBucket: "pto-app-30271.appspot.com",
  messagingSenderId: "440515616929",
  appId: "1:440515616929:ios:1be65de5cfa7a0048210e0",
  measurementId: "1:440515616929:ios:1be65de5cfa7a0048210e0",
};

const FIREBASE_APP = initializeApp(firebaseConfig);

const FIREBASE_AUTH = getAuth(FIREBASE_APP);
const FIRESTORE_DB = getFirestore(FIREBASE_APP);

if (process.env.NODE_ENV === "test") {
  console.log("connecting to emulator");
  connectAuthEmulator(FIREBASE_AUTH, "http://127.0.0.1:9099");
  connectFirestoreEmulator(FIRESTORE_DB, "http://127.0.0.1", 8080);
}

export { FIREBASE_APP, FIREBASE_AUTH, FIRESTORE_DB };
