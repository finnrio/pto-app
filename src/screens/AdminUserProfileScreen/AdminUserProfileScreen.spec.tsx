import { cleanup, render } from "@testing-library/react-native";
import React from "react";
import AdminUserProfileScreen from "./AdminUserProfileScreen";
import { UserContext } from "../../context/UserContext";
import { User } from "firebase/auth";
import { createMock } from "@golevelup/ts-jest";

// solution from https://github.com/APSL/react-native-keyboard-aware-scroll-view/issues/493#issuecomment-1023551697
jest.mock("react-native-keyboard-aware-scroll-view", () => ({
  KeyboardAwareScrollView: (props: { children: React.ReactNode }) =>
    props.children,
}));

// mock the GetAllUsers function
jest.mock("../../firebase/operations/GetAllUsers", () => {
  return jest.fn(() => {
    return Promise.resolve([
      {
        id: "1",
        first_name: "User",
        surname: "One",
        email: "test@test.test",
        role: "User",
      },
      {
        id: "2",
        first_name: "User",
        surname: "Two",
        email: "test@test.test",
        role: "User",
      },
    ]);
  });
});

// mock the GetAllManagers function
jest.mock("../../firebase/operations/GetAllManagers", () => {
  return jest.fn(() => {
    return Promise.resolve([
      {
        id: "3",
        first_name: "Manager",
        surname: "One",
        email: "test@test.test",
        role: "Manager",
      },
      {
        id: "4",
        first_name: "Manager",
        surname: "Two",
        email: "test@test.test",
        role: "Manager",
      },
    ]);
  });
});

jest.mock("../../firebase/operations/GetUserData", () => {
  return jest.fn(() => {
    return Promise.resolve({
      first_name: "mock_first_name",
      surname: "mock_surname",
      email: "mock_email",
      role: "mock_role",
    });
  });
});

const mockAuthUser: User = createMock<User>({ uid: "mock_uid" });

describe("AdminUserProfileScreen", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const { toJSON } = render(
      <UserContext.Provider value={mockAuthUser}>
        <AdminUserProfileScreen />
      </UserContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  // describe("when a user is selected", () => {
  // it("renders the user's data", async () => {
  //   // await act(async () => {
  //     render(<AdminUserProfileScreen />);
  //   // });
  //   const hook = renderHook(() => AdminUserProfileScreen());
  //   // await waitFor(() => expect(hook.result.current.allUsersList).toBe("User One"));
  //   // await act(async () => {
  //   //   await fireEvent.press(screen.getByText("Select a User"));
  //   // });
  //   // await act(async () => {
  //   //   await fireEvent.press(await screen.getByText("User One"));
  //   // });
  //   // get the data in the selected user SelectList component
  //   // await act(() => {
  //     // fireEvent.press(screen.getByTestId("user_dropdown").props.value);
  //   // });
  //   console.log(screen.getByTestId("user_dropdown").props.value);
  //   // await act(() => {
  //   //   fireEvent.press(screen.getByTestId("User One"));
  //   // });
  //   // await screen.getByTestId("user_dropdown").props.setValue("User One");
  //   // fireEvent.press(screen.getByTestId("User One"));
  //   // await fireEvent(userDropdown, 'onValueChange', 'User One');
  //   // console.log(screen.getByTestId("user_dropdown").children[0]);
  //   // console.log(screen.getByTestId("user_dropdown").children[0]);
  //   expect(screen.getByTestId("first_name_input")).toBeTruthy();
  //   // expect(screen.getByTestId("surname_input")).toBeTruthy();
  //   // expect(screen.getByTestId("email_input")).toBeTruthy();
  //   // expect(screen.getByTestId("role_input")).toBeTruthy();
  //   // expect(screen.getByTestId("update_button")).toBeTruthy();
  //   // expect(screen.getByTestId("delete_button")).toBeTruthy();
  // });
  // });
});
