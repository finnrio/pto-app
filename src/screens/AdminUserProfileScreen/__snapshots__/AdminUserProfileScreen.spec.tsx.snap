// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUserProfileScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "zIndex": 5000,
        },
        {},
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 50,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 3,
          "width": "100%",
          "zIndex": 5000,
        }
      }
      testID="user_dropdown"
    >
      <Text
        style={
          [
            {
              "color": "#000",
              "flex": 1,
            },
            {},
            false,
            {},
          ]
        }
      >
        Select an item
      </Text>
      <View
        style={
          [
            {
              "marginLeft": 10,
            },
            {},
          ]
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "height": 20,
                "width": 20,
              },
              {},
            ]
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    User ID
  </Text>
  <TextInput
    editable={false}
    placeholder="No user selected"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="uid_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    First Name
  </Text>
  <TextInput
    autoCapitalize="none"
    editable={false}
    onChangeText={[Function]}
    placeholder="First Name"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="firstName_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Surname
  </Text>
  <TextInput
    autoCapitalize="none"
    editable={false}
    onChangeText={[Function]}
    placeholder="Surname"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="surname_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Email
  </Text>
  <TextInput
    autoCapitalize="none"
    editable={false}
    onChangeText={[Function]}
    placeholder="E-mail"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="email_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Role
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderBottomRightRadius": 0,
            "borderColor": "rgba(28, 27, 31, 0.12)",
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
            "borderWidth": 1,
          },
          {
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 76,
          },
          [
            undefined,
            {},
          ],
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderBottomRightRadius": 0,
              "borderRadius": 20,
              "borderRightWidth": 0,
              "borderTopRightRadius": 0,
            },
          ]
        }
        testID="user_button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 9,
              },
              {
                "paddingVertical": 9,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ]
            }
            testID="user_button-label"
          >
            User
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderBottomLeftRadius": 0,
            "borderColor": "rgba(28, 27, 31, 0.12)",
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
            "borderWidth": 1,
          },
          {
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 76,
          },
          [
            undefined,
            {},
          ],
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderBottomLeftRadius": 0,
              "borderRadius": 20,
              "borderTopLeftRadius": 0,
            },
          ]
        }
        testID="manager_button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 16,
                "paddingVertical": 9,
              },
              {
                "paddingVertical": 9,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "textAlign": "center",
                    },
                    {
                      "color": "rgba(28, 27, 31, 0.38)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ]
            }
            testID="manager_button-label"
          >
            Manager
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Manager
  </Text>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "zIndex": 5000,
        },
        {},
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 50,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 3,
          "width": "100%",
          "zIndex": 5000,
        }
      }
      testID="manager_dropdown"
    >
      <Text
        style={
          [
            {
              "color": "#000",
              "flex": 1,
            },
            {},
            false,
            {},
          ]
        }
      >
        Select an item
      </Text>
      <View
        style={
          [
            {
              "marginLeft": 10,
            },
            {},
          ]
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "height": 20,
                "width": 20,
              },
              {},
            ]
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    PTO Allowance
  </Text>
  <TextInput
    autoCapitalize="none"
    editable={false}
    onChangeText={[Function]}
    placeholder="PTO Allowance"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="pto_allowance_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    PTO Used
  </Text>
  <TextInput
    autoCapitalize="none"
    editable={false}
    onChangeText={[Function]}
    placeholder="PTO Used"
    placeholderTextColor="#aaaaaa"
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 5,
        "height": 48,
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "overflow": "hidden",
        "paddingLeft": 16,
      }
    }
    testID="pto_used_input"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="update_button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Update Profile
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="remove_button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Remove User
    </Text>
  </View>
</View>
`;
